src/bun.js/api/bun/dns_resolver.zig

```diff
@@ -2376,6 +2583,94 @@ pub const DNSResolver = struct {
         return values;
     }
 
+    // Resolves the given address and port into a host name and service using the operating system's underlying getnameinfo implementation.
+    // If address is not a valid IP address, a TypeError will be thrown. The port will be coerced to a number.
+    // If it is not a legal port, a TypeError will be thrown.
+    pub fn lookupService(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSC.JSValue {
+        const arguments = callframe.arguments(3);
+        if (arguments.len < 2) {
+            globalThis.throwNotEnoughArguments("lookupService", 3, arguments.len);
+            return .zero;
+        }
+
+        const addr_value = arguments.ptr[0];
+        const port_value = arguments.ptr[1];
+        if (addr_value.isEmptyOrUndefinedOrNull() or !addr_value.isString()) {
+            globalThis.throwInvalidArgumentType("lookupService", "address", "string");
+            return .zero;
+        }
+        const addr_str = addr_value.toStringOrNull(globalThis) orelse {
+            return .zero;
+        };
+        if (addr_str.length() == 0) {
+            globalThis.throwInvalidArgumentType("lookupService", "address", "non-empty string");
+            return .zero;
+        }
+
+        const addr_s = addr_str.getZigString(globalThis).slice();
+        const port: u16 = if (port_value.isNumber()) blk: {
+            break :blk port_value.to(u16);
+        } else {
+            globalThis.throwInvalidArgumentType("lookupService", "port", "invalid port");
+            return .zero;
+        };
+
+        const addr = std.net.Address.parseIp(addr_s, port) catch {
```

Comment:
Is there a c-ares function we can use for parsing ip addresses? I am not confident in the quality reliability or security of `std.net.Address.parse*`

Reply:
Fixed. Added a `getSockaddr` function to get the `sockaddr` struct, which uses `ares_inet_pton` to parse the address.